### ISSUE TITLE
Feature request: Add columns for total resources for process tree

### ISSUE THREAD

**Issue by svenssonaxel** [htop-dev/htop#1154]
Currently, tree-view is almost useless for finding resource hogs. The reason is that sorting on e.g. CPU use in tree-view mode, will sort the process tree based on the CPU use of the parent, rather than the total.

Adding columns for total CPU/memory use for a process tree would allow the user to sort on the total to surface the process trees that hog the most resources, yet retain the ability to inspect the constituent processes and threads individually.

It is possible that users might want to sort on total CPU for the tree but without showing the column. Therefore, a further improvement would be to add an option that, in tree-mode, sorting on resource columns will sort on the total for the tree.

There are a couple of suggestions and PRs related to total resources. Here's how this one differs:
* This is different from #920 since sorting process trees will need the total no matter if they're collapsed or expanded.
* This is different from #301 since 1) it is defined exactly rather than reliant on a heuristic and 2) it retains the ability to inspect the process tree of an application.

**Comment by FranklinYu** [comment]
Do we simply sum up all the CPU and RAM usage? Is there a case where there is some overlap, e.g. “shared memory”?

**Comment by BenBE** [comment]
Yes, e.g. memory used for shared libraries/executables. Also virtual memory may be shared between processes due to forking and a partially shared memory map.
